tests/unit/lms/services/jstor/factory_test.py

```diff
@@ -35,9 +23,54 @@ def test_it(
             enabled=sentinel.jstor_enabled,
             site_code=sentinel.jstor_site_code,
             http_service=http_service,
+            tracking_user_agent="Chrome 100",
+            tracking_user_id=pyramid_request.lti_user.h_user.username,
         )
         assert svc == JSTORService.return_value
 
+    def test_it_omits_tracking_info_if_not_available(
+        self, pyramid_request, http_service, JSTORService
+    ):
+        pyramid_request.registry.settings.update(
```

Comment:
I tried to put this statement in the shared `pyramid_request` fixture but ran into an issue where the `settings` object was not available at that time:

```
    @pytest.fixture(autouse=True)
    def pyramid_request(self, pyramid_request):
>       pyramid_request.registry.settings.update(
            {
                "jstor_api_url": sentinel.jstor_api_url,
                "jstor_api_secret": sentinel.jstor_api_secret,
            }
        )
E       AttributeError: 'NoneType' object has no attribute 'update'
```

I guess this may be an issue with the order in which overridden fixtures are run. Any suggestions on how best to fix it?